packages/protocol-http/src/Fields.ts

```diff
@@ -0,0 +1,39 @@
+import { Field } from "./Field";
+import { FieldPosition } from "./FieldPosition";
+
+export type FieldsOptions = { fields?: Field[]; encoding?: string; };
+
+/**
+ * Collection of Field entries mapped by name.
+ */
+export class Fields {
+  private readonly entries: Record<string, Field> = {};
+  private readonly encoding: string;
+
+  constructor({ fields = [], encoding = "utf-8" }: FieldsOptions) {
+    fields.forEach(this.setField.bind(this));
+    this.encoding = encoding;
+  }
+
+  // Set entry for a Field name. the `name`
+  // attribute will be used to key the collection.
+  public setField(field: Field): void {
+    this.entries[field.name] = field;
+  }
+
+  // Retrieve Field entry by name.
+  public getField(name: string): Field | undefined {
+    return this.entries[name];
+  }
+
+  // Delete entry from collection.
+  public removeField(name: string): void {
+    delete this.entries[name];
+  }
+
+  // Helper function for retrieving specific types of fields.
+  // Used to grab all headers or all trailers.
```

Comment:
not critical for this PR, but please use the jsdoc / tsdoc format for code comments on methods.

Reply:
I've updated this PR to use jsdoc / tsdoc format to the best of my knowledge. Let me know how it looks.